crates/tabby-index/src/indexer.rs

```diff
@@ -72,21 +72,39 @@ impl<T: ToIndexId> TantivyDocBuilder<T> {
         let now = tantivy::time::OffsetDateTime::now_utc();
         let updated_at = tantivy::DateTime::from_utc(now);
 
-        let doc = doc! {
-            schema.field_id => id,
-            schema.field_source_id => source_id,
-            schema.field_corpus => self.corpus,
-            schema.field_attributes => self.builder.build_attributes(&document).await,
-            schema.field_updated_at => updated_at,
-        };
-
         let cloned_id = id.clone();
+        let doc_id = id.clone();
+        let doc_attributes = self.builder.build_attributes(&document).await;
         let s = stream! {
-            yield tokio::spawn(async move { Some(doc) });
+            let mut failed_count: u64 = 0;
+            for await chunk_doc in self.build_chunks(cloned_id, source_id.clone(), updated_at, document).await {
+                match chunk_doc.await {
+                    Ok((Some(doc), ok)) => {
+                        if !ok {
+                            failed_count += 1;
+                        }
+                        yield tokio::spawn(async move { Some(doc) });
+                    }
+                    Ok((None, _)) => {
+                        failed_count += 1;
+                    }
+                    Err(e) => {
+                        warn!("Failed to build chunk for document '{}': {}", doc_id, e);
+                        failed_count += 1;
+                    }
+                }
+            };
 
-            for await doc in self.build_chunks(cloned_id, source_id, updated_at, document).await {
-                yield doc;
-            }
+            let doc = doc! {
+                schema.field_id => doc_id,
+                schema.field_source_id => source_id,
+                schema.field_corpus => self.corpus,
+                schema.field_attributes => doc_attributes,
+                schema.field_updated_at => updated_at,
+                schema.field_failed_chunks_count => failed_count,
```

Comment:
you might only set failed_chunks_count when failed_count > 0, so the search query can be simplified slightly.